src/viewport.js

```diff
@@ -379,59 +390,46 @@ export class Viewport {
   /** @private */
   scroll_() {
     this.scrollCount_++;
-    this.scrollObservable_.fire();
-
-    this.scrollLeft_ = this.binding_.getScrollLeft();
-
-    if (this.scrollTracking_) {
-      return;
-    }
-
-    const newScrollTop = this.binding_.getScrollTop();
-    if (newScrollTop < 0) {
-      // iOS and some other browsers use negative values of scrollTop for
-      // overscroll. Overscroll does not affect the viewport and thus should
-      // be ignored here.
-      return;
-    }
-
-    this.scrollTracking_ = true;
-    this.scrollTop_ = newScrollTop;
-    this.scrollMeasureTime_ = timer.now();
-    timer.delay(() => this.scrollDeferred_(), 500);
+    this.vsync_.measure(() => {
```

Comment:
Good point. Moved out of vsync.

Actually, if you only measure and do not mutate, it is always fine to do that in an event handler. In the worst case it trigger recalc but it would make the next vsync cheaper since we start with the measure phase.